letsencrypt-auto-source/pieces/bootstrappers/rpm_python3.sh

```diff
@@ -0,0 +1,25 @@
+# If new packages are installed by BootstrapRpmPython3 below, this version
+# number must be increased.
+BOOTSTRAP_RPM_PYTHON3_VERSION=1
+
+BootstrapRpmPython3() {
+  # Tested with:
+  #   - CentOS 6 (EPEL must be installed manually)
+
+  FindInstallTool
+  tool=$?
+
+  # EPEL uses python34
+  if $tool list python34 >/dev/null 2>&1; then
+    python_pkgs="python34
+      python34-devel
+      python-virtualenv
+      python34-tools
+      python-pip
+    "
+  fi
+  # TODO: add some elifs and elses for other distros that
+  # might have 2.6 or no python installed to get here
```

Comment:
Yes, adding an else seems like the right call.

I don't have access yet, I'd love those stats. And also if you have any easier ways to check package lists on those systems than installing each one with docker.

Reply:
I'm pulling the stats now for distinct `certbot-auto` installs in the past 2 months using Python 2.6. I'll post results here when the query is done.

Unfortunately I really don't have a better way to query versions for Red Hat based systems. All browser search tools I've seen are really buggy/inaccurate. If persistence is at all useful, I think you should feel free to spin up droplets/EC2 instances while you're working on this issue.

Reply:
Here's the count of distinct IPs using Python 2.6 and `certbot-auto` by OS.

I've only included OSes with at least 5 users as below that, some of the values get pretty specific and start to identify the user. There are 61 more different OS strings with less than 5 users.

Also, keep in mind that we don't officially support many of the OSes below and the user had to do some tinkering on their end to make things work.

| "os"                                  | "_count_distinct" | 
|---------------------------------------|-------------------| 
| "CentOS 6.9"                          | "22480"           | 
| "CentOS 6.8"                          | "8180"            | 
| "CentOS 6.7"                          | "3236"            | 
| "CentOS 6.5"                          | "2885"            | 
| "CentOS 6.6"                          | "2088"            | 
| "CentOS 6.4"                          | "815"             | 
| "Red Hat Enterprise Linux Server 6.9" | "455"             | 
| "CentOS 6.3"                          | "390"             | 
| "CentOS 6.2"                          | "220"             | 
| "Scientific Linux 6.9"                | "160"             | 
| "Oracle Linux Server 6.9"             | "108"             | 
| "CentOS Linux 6.0"                    | "83"              | 
| "Red Hat Enterprise Linux Server 6.8" | "73"              | 
| "Scientific Linux 6.8"                | "64"              | 
| "Oracle Linux Server 6.8"             | "53"              | 
| "Red Hat Enterprise Linux Server 6.7" | "51"              | 
| "Red Hat Enterprise Linux Server 6.6" | "47"              | 
| "Red Hat Enterprise Linux Server 6.5" | "45"              | 
| "SHMZ 6.6"                            | "42"              | 
| "CloudLinux Server 6.9"               | "31"              | 
| "Red Hat Enterprise Linux Server 6.4" | "29"              | 
| "SHMZ 6.5"                            | "23"              | 
| "ClearOS Community 6.9.0"             | "21"              | 
| "Scientific Linux 6.5"                | "18"              | 
| "debian 6.0.10"                       | "17"              | 
| "Oracle Linux Server 6.7"             | "15"              | 
| "CentOS 6.1"                          | "15"              | 
| "CentOS 5.11"                         | "14"              | 
| "Scientific Linux 6.7"                | "14"              | 
| "Oracle Linux Server 6.6"             | "11"              | 
| "Scientific Linux 6.1"                | "11"              | 
| "Red Hat Enterprise Linux Server 6.2" | "10"              | 
| "Ubuntu 10.04"                        | "10"              | 
| "Scientific Linux 6.4"                | "10"              | 
| "Scientific Linux 6.6"                | "10"              | 
| "Red Hat Enterprise Linux Server 6.3" | "9"               | 
| "Amazon Linux AMI 2014.09"            | "8"               | 
| "Oracle Linux Server 6.4"             | "8"               | 
| "Scientific Linux 6.3"                | "7"               | 
| "Scientific Linux CERN SLC 6.9"       | "7"               | 
| "Scientific Linux 6.2"                | "6"               | 
| "Oracle Linux Server 6.5"             | "6"               | 
| "Red Hat Enterprise Linux Server 6.0" | "6"               | 
| "Scientific Linux 6.0"                | "6"               | 
| "Red Hat Enterprise Linux Server 6.1" | "5"               | 
| "CentOS 5.8"                          | "5"               |

Reply:
As I understand it, using `epel-release` means package names are consistent, right? They're all pulling down the same package list once they have that? And they'll all have to, since everything here that we support that's redhat is a version of RHEL 6, essentially.

Reply:
To be a little pedantic to share some background and make sure we're on the same page, `yum install epel-release` installs a CentOS specific package that automates setting up the [EPEL](https://fedoraproject.org/wiki/EPEL) repository on that system. It doesn't work on RHEL for example.

With that said, yes, the package names and contents in EPEL 6 are the same for all the RHEL 6 variants supported by the repo.